source/synthDriverHandler.py

```diff
@@ -110,12 +116,19 @@ def setSynth(name,isFallback=False):
 		return True
 	except:
 		log.error("setSynth", exc_info=True)
+		# As there was an error loading this synth:
 		if prevSynthName:
+			# There was a previous synthesizer, so switch back to that one. 
 			setSynth(prevSynthName,isFallback=True)
-		elif name not in ('espeak','silence'):
-			setSynth('espeak',isFallback=True)
-		elif name=='espeak':
-			setSynth('silence',isFallback=True)
+		else:
+			# There was no previous synth, so fallback to the next available default synthesizer that has not been tried yet.
+			try:
+				nextIndex=defaultSynthPriorityList.index(name)+1
+			except ValueError:
+				nextIndex=0
+			if nextIndex<len(defaultSynthPriorityList):

```

Comment:
In the very unlikely case that the silence synthesizer does not load, will this create an endless loop? I.e.

1. OneCore doesn't load, go to espeak
2. Espeak doesn't load, go to silence
3. Silence doesn't load, index is 0, go to OneCore

I guess that if nextIndex==len(defaultSynthPriorityList) means that we're stuck with regard to speech. Should we account for this?